tools/clang/include/clang/SPIRV/SpirvInstruction.h

```diff
@@ -381,14 +384,20 @@ class SpirvFunctionParameter : public SpirvInstruction {
   }
 };
 
+/// \brief Represents an OpLabel instruction
+class SpirvLabel : public SpirvInstruction {
+public:
+  SpirvLabel(uint32_t resultId, SourceLocation loc);
+};
+
 /// \brief Merge instructions include OpLoopMerge and OpSelectionMerge
 class SpirvMerge : public SpirvInstruction {
 public:
-  uint32_t getMergeBlock() const { return mergeBlock; }
+  SpirvLabel *getMergeBlock() const { return mergeBlock; }
```

Comment:
I think this should be `SpirvBasicBlock` instead of introducing a new `SpirvLabel` type?